common/lib/xmodule/xmodule/js/src/video/09_completion.js

```diff
@@ -0,0 +1,173 @@
+                }
+                if (this.lastSentTime !== undefined && currentTime - this.lastSentTime < this.repostDelayTime()) {
+                    // Throttle attempts to submit in case of network issues
+                    return;
+                }
+                if (this.completeAfterTime === undefined) {
+                    // Duration is not available at initialization time
+                    duration = this.state.videoPlayer.duration();
+                    if (!duration) {
+                        // duration is not yet set. Wait for another event,
+                        // or fall back to 'ended' handler.
+                        return;
+                    }
+                    this.completeAfterTime = this.calculateCompleteAfterTime(this.startTime, duration);
+                }
+
+                if (currentTime > this.completeAfterTime) {
+                    this.markCompletion(currentTime);
+                }
+            },
+
+            /** Submit completion to the LMS */
+            markCompletion: function(currentTime) {
+                var self = this;
+                var errmsg;
+                this.isComplete = true;
+                this.lastSentTime = currentTime;
+                if (this.state.config.publishCompletionUrl) {
+                    $.ajax({
+                        type: 'POST',
+                        url: this.state.config.publishCompletionUrl,
+                        contentType: 'application/json',
+                        dataType: 'json',
+                        data: JSON.stringify({completion: 1.0}),
+                        success: function() {
+                            self.state.el.off('timeupdate.completion');
+                            self.state.el.off('ended.completion');
+                        },
+                        error: function(xhr) {
+                            /* eslint-disable no-console */
+                            self.complete = false;
+                            errmsg = 'Failed to submit completion';
+                            if (xhr.responseJSON !== undefined) {
+                                errmsg += ': ' + xhr.responseJSON.error;
+                            }
+                            console.warn(errmsg);
+                            /* eslint-enable no-console */
+                        }
+                    });
+                } else {
+                    /* eslint-disable no-console */
+                    console.warn('publishCompletionUrl not defined');
+                    /* eslint-enable no-console */
+                }
+            },
+
+            /** Determine what point in the video counts as complete. */
+            calculateCompleteAfterTime: function(startTime, endTime) {
+                return startTime + (endTime - startTime) * this.completionPercentage;
+            },
+
+            /** How many seconds to wait after a POST fails to try again. */
+            repostDelayTime: function() {
```

Comment:
I actually would have left this as `repostDelaySeconds`, since this is not seconds from the beginning of the video like all the other Time variables.  I think it would be more consistent if this went back as it was.

Reply:
Makes sense.